modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/DXPActivationKeysTable/index.js

```diff
@@ -0,0 +1,458 @@
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayIcon from '@clayui/icon';
+import ClaySticker from '@clayui/sticker';
+import {ClayTooltipProvider} from '@clayui/tooltip';
+import {useEffect, useMemo, useState} from 'react';
+import {IconButton} from '../../../../common/components';
+import RoundedGroupButtons from '../../../../common/components/RoundedGroupButtons';
+import Table from '../../../../common/components/Table';
+import {useApplicationProvider} from '../../../../common/context/AppPropertiesProvider';
+import {
+	getActivationDownloadKey,
+	getActivationLicenseKey,
+} from '../../../../common/services/liferay/rest/raysource/LicenseKeys';
+import downloadFromBlob from '../../../../common/utils/downloadFromBlob';
+import getCurrentEndDate from '../../../../common/utils/getCurrentEndDate';
+import {useCustomerPortal} from '../../context';
+import {EXTENSION_FILE_TYPES, STATUS_CODE} from '../../utils/constants';
+import {getPascalCase} from '../../utils/getPascalCase';
+
+const ACTIVATION_KEYS_LICENSE_FILTER_TYPES = {
+	active: ({expirationDate, startDate}) =>
+		new Date(startDate) < new Date() &&
+		new Date(expirationDate) > new Date(),
+	expired: ({expirationDate}) => new Date(expirationDate) < new Date(),
+	notActivated: ({startDate}) => new Date(startDate) > new Date(),
+};
+
+const ACTIVATION_STATUS = {
+	active: {
+		color: 'success',
+		id: 'active',
+		title: 'Active',
+	},
+	all: {
+		color: 'none',
+		id: 'all',
+		title: 'All',
+	},
+	expired: {
+		color: 'danger',
+		id: 'expired',
+		title: 'Expired',
+	},
+	notActivated: {
+		color: 'info',
+		id: 'notActivated',
+		title: 'Not Activated',
+	},
+};
+
+const VIRTUAL_CLUSTER = 'virtual-cluster';
+
+const DXPActivationKeysTable = () => {
+	const [{assetsPath, project, sessionId}] = useCustomerPortal();
+	const {licenseKeyDownloadURL} = useApplicationProvider();
+
+	const [activationKeys, setActivationKeys] = useState({data: []});
+	const [activationKeysFiltered, setActivationKeysFiltered] = useState([]);
+	const [totalCount, setTotalCount] = useState(5);
+
+	const [filterStatusBar, setFilterStatusBar] = useState('all');
+
+	const [activePage, setActivePage] = useState(1);
+	const [itemsPerPage, setItemsPerPage] = useState(5);
+	const [isLoadingActivationKeys, setIsLoadingActivationKeys] = useState(
+		false
+	);
+
+	useEffect(() => {
+		if (filterStatusBar) {
+			setActivePage(1);
+		}
+	}, [filterStatusBar]);
+
+	useEffect(() => {
+		setIsLoadingActivationKeys(true);
+		const fetchActivationKeysData = async () => {
+			const {items} = await getActivationLicenseKey(
+				project.accountKey,
+				licenseKeyDownloadURL,
+				encodeURI('active eq true'),
+				1,
+				9999,
```

Comment:
I didn't get this, I think that will be more clear if use declarative constants instead "magic numbers", makes sense?

Reply:
Indeed! I'll be adding that